src/main/java/org/springframework/cloud/scheduler/spi/cloudfoundry/CloudFoundryAppScheduler.java

```diff
@@ -134,18 +193,155 @@ private String stageTask(ScheduleRequest scheduleRequest) {
+	 * @return {@link Flux} containing the {@link ScheduleInfo}s for the specified page number.
+	 */
+	private Flux<ScheduleInfo> getSchedules(int pageNumber) {
+		Flux<ApplicationSummary> applicationSummaries = cacheAppSummaries();
+		return this.getSpace(this.properties.getSpace()).flatMap(requestSummary -> {
+			return this.client.jobs().list(ListJobsRequest.builder()
+					.spaceId(requestSummary.getId())
+					.page(pageNumber)
+					.detailed(true).build()); })
+				.flatMapIterable
+						(jobs -> jobs.getResources())// iterate over the resources returned.
+				.flatMap(job -> {
+					return getApplication(applicationSummaries,
+							job.getApplicationId()) // get the application name for each job.
+							.map(optionalApp -> {
+								ScheduleInfo scheduleJobInfo = new ScheduleInfo();
+								scheduleJobInfo.setScheduleProperties(new HashMap<>());
+								scheduleJobInfo.setScheduleName(job.getName());
+								scheduleJobInfo.setTaskDefinitionName(optionalApp.getName());
+								if (job.getJobSchedules().isPresent()) {
+									scheduleJobInfo.getScheduleProperties().put(SchedulerPropertyKeys.CRON_EXPRESSION,
+											job.getJobSchedules().get().get(0).getExpression());
+								}
+								else {
+									logger.warn(String.format("Job %s does not have an associated schedule", job.getName()));
+								}
+								return scheduleJobInfo;
+							});
+				});
+	}
+
+	/**
+	 * Retrieves the number of pages that can be returned when retrieving a list of jobs.
+	 * @return an int containing the number of available pages.
+	 */
+	private int getJobPageCount() {
+		ListJobsResponse response = this.getSpace(this.properties.getSpace()).flatMap(requestSummary -> {
+			return this.client.jobs().list(ListJobsRequest.builder()
+					.spaceId(requestSummary.getId())
+					.detailed(true).build());
+		}).block();
+		return response.getPagination().getTotalPages();
+	}
+
+	/**
+	 * Retrieve a {@link Mono} that contains the {@link Job} for the jobName or null.
+	 * @param jobName - the name of the job to search search.
+	 * @param page - the page to search.
+	 * @return {@link Mono} containing the {@link Job} if found or null if not found.
+	 */
+	private Mono<Job> getJobMono(String jobName, int page) {
+		return this.getSpace(this.properties.getSpace()).flatMap(requestSummary -> {
+			return this.client
+					.jobs()
+					.list(ListJobsRequest.builder()
+							.spaceId(requestSummary.getId())
+							.page(page)
+							.build()); })
+				.flatMapIterable(jobs -> jobs.getResources())
+				.filter(job ->
+						job.getName().equals(jobName))
+				.singleOrEmpty();// iterate over the resources returned.
+	}
+
+	/**
+	 * Retrieve the job id for the specified PCF Job Name.
+	 * @param jobName the name of the job to search.
+	 * @return The job id associated with the job.
+	 */
+	private String getJob(String jobName) {
+		Job result = null;
+
+		for (int pageNum = PCF_PAGE_START_NUM; pageNum <= getJobPageCount(); pageNum++) {
```

Comment:
This is going to call `getJobPageCount()` on each iteration of the loop which issues a number of REST calls.  I'd recommend making this call once.

Reply:
done